django/core/serializers/python.py

```diff
@@ -34,8 +34,32 @@ def end_object(self, obj):
 
     def get_dump_object(self, obj):
         data = {"model": str(obj._meta)}
-        if not self.use_natural_primary_keys or not hasattr(obj, "natural_key"):
+
+        pk_included = True
+
+        if self.use_natural_primary_keys:
+            natural_key_func = getattr(obj, "natural_key", None)
+
+            if callable(natural_key_func):
+                natural_key_value = None
+                try:
+                    natural_key_value = natural_key_func()
+                except Exception:
+                    pass
+
+                is_opt_out = (
+                    natural_key_value is None
+                    or not natural_key_value
+                    or natural_key_value == (obj.pk,)
```

Comment:
A 1-tuple of just `"pk"` feels like a little bit of a hack / contradiction. If we don't add support for this, then we can just use implicit booleanness (truthiness).

Reply:
Got it, that makes sense.
I’ll keep the current None handling for this PR, and we can revisit its removal in a follow-up change if needed!